crates/codegen/syntax_templates/src/shared/language.rs

```diff
@@ -131,8 +158,8 @@ pub(crate) fn render_error_report(
 
     let kind = ReportKind::Error;
     let color = if with_color { Color::Red } else { Color::Unset };
-    let source_start = error.position;
-    let source_end = error.position;
+    let source_start = error.position.char;
```

Comment:
IIUC, these are also used to create the `ariadne` span, which now we send chars instead of bytes. Should we keep bytes here?

Reply:
ariadne is user facing, so it should be chars and not bytes

Reply:
`source_start` and `source_end` are used in two places:

1. In the user facing display strings: which is fine to use char offsets.
2. In constructing `Label` and `Report` offsets, which expect bytes (?).

I suspect this change can break it? I don't think we have tests that cover the second case, but we can probably add one.

Reply:
In the label offsets it gets shown to the user, so it should also be char offsets.